src/infra/config/dex/okx/file.rs

```diff
@@ -0,0 +1,63 @@
+use {
+    crate::{
+        domain::eth,
+        infra::{config::dex::file, contracts, dex::okx},
+        util::serialize,
+    },
+    serde::Deserialize,
+    serde_with::serde_as,
+    std::path::Path,
+};
+
+#[serde_as]
+#[derive(Deserialize)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields)]
+struct Config {
+    /// The versioned URL endpoint for the 0x swap API.
+    #[serde(default = "default_endpoint")]
+    #[serde_as(as = "serde_with::DisplayFromStr")]
+    endpoint: reqwest::Url,
+
+    /// Chain ID used to automatically determine contract addresses.
+    #[serde_as(as = "serialize::ChainId")]
+    chain_id: eth::ChainId,
+
+    pub api_project_id: String,
+
+    pub api_key: String,
+
+    pub api_secret_key: String,
+
+    pub api_passphrase: String,
```

Comment:
Could be grouped into a separate struct since the parameters are closely related.